packages/react-dom/src/client/focus/__tests__/TabFocusController-test.internal.js

```diff
@@ -221,5 +215,44 @@ describe('TabFocusContainer', () => {
       createEventTarget(document.activeElement).tabPrevious();
       expect(document.activeElement).toBe(button2Ref.current);
     });
+
+    it('allows for imperative tab focus control', () => {
+      const firstFocusControllerRef = React.createRef();
+      const secondFocusControllerRef = React.createRef();
+      const buttonRef = React.createRef();
+      const button2Ref = React.createRef();
+      const divRef = React.createRef();
+
+      const Test = () => (
+        <div>
+          <TabFocusController ref={firstFocusControllerRef}>
+            <input tabIndex={-1} />
+            <button ref={buttonRef} />
+            <button ref={button2Ref} />
+            <input tabIndex={-1} />
+          </TabFocusController>
+          <TabFocusController ref={secondFocusControllerRef}>
+            <input tabIndex={-1} />
+            <div ref={divRef} tabIndex={0} />
+          </TabFocusController>
+        </div>
+      );
+
+      ReactDOM.render(<Test />, container);
+      firstFocusControllerRef.current.focusFirst();
```

Comment:
The tests are a bit easier to read if we avoid writing out `ref.current` each time:

```
const button = buttonRef.current;
const firstFocusController = firstFocusControllerRef.current;

firstFocusController.focusFirst();
expect(document.activeElement).toBe(button);
firstFocusController.focusNext();
expect(document.activeElement).toBe(button2);
```